core/templates/pages/contributor-dashboard-page/contributor-badge/contributor-badge.component.ts

```diff
@@ -0,0 +1,395 @@
+  mobileBadgeTypeDropdownShown = false;
+  mobileTranslationBadgesShown = true;
+  selectedLanguage = '';
+  dataLoading = false;
+  userCanReviewTranslationSuggestion: boolean;
+  userCanReviewQuestionSuggestions: boolean;
+  userCanSuggestQuestions: boolean;
+  reviewableLanguages = [];
+  constructor(
+    private readonly languageUtilService: LanguageUtilService,
+    private readonly contributionAndReviewStatsService:
+      ContributionAndReviewStatsService,
+    private readonly userService: UserService) {}
+
+  async ngOnInit(): Promise<void> {
+    this.dataLoading = true;
+    const userInfo = await this.userService.getUserInfoAsync();
+    const username = userInfo.getUsername();
+
+    if (username === null) {
+      throw new Error('Cannot fetch username.');
+    }
+
+    const userContributionRights =
+      await this.userService.getUserContributionRightsDataAsync();
+    userContributionRights.can_review_translation_for_language_codes.map(
+      (code) => {
+        const languageDescription = this.languageUtilService
+          .getAudioLanguageDescription(
+            code);
+        this.totalTranslationStats[languageDescription] = {
+          language: this.languageUtilService.getShortLanguageDescription(
+            languageDescription),
+          submissions: 0,
+          reviews: 0,
+          corrections: 0,
+          allSubmissionBadgesCreated: false,
+          allReviewBadgesCreated: false,
+          allCorrectionBadgesCreated: false
+        };
+        this.reviewableLanguages.push(languageDescription);
+      });
+    this.userCanReviewQuestionSuggestions = (
+      userContributionRights.can_review_questions);
+    this.userCanSuggestQuestions = (
+      userContributionRights.can_suggest_questions);
+
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      await response.translation_contribution_stats.map((stat) => {
+        const languageDescription =
+          this.languageUtilService.getAudioLanguageDescription(
+            stat.language_code);
+
+        // There can be languages that the contributor has translation
+        // contribution stats but not translation review stats. Hence we need to
+        // initialize TotalTranslationStats objects for those languages.
+        if (!this.totalTranslationStats[languageDescription]) {
+          this.totalTranslationStats[languageDescription] = {
+            language: this.languageUtilService.getShortLanguageDescription(
+              languageDescription),
+            submissions: stat.accepted_translations_count,
+            reviews: 0,
+            corrections: 0,
+            allSubmissionBadgesCreated: false,
+            allReviewBadgesCreated: false,
+            allCorrectionBadgesCreated: false
+          };
+        } else {
+          this.totalTranslationStats[languageDescription].submissions += (
```

Comment:
```suggestion
          this.totalTranslationStats[languageDescription].submissions = (
```

I don't think you need to add the value, just set it. Here and below.

Reply:
Here one stat record gives the number of submissions for a given topic in a given language. If we want to get the total submissions for the language we need to get the sum of contributions for all topics so we need to add the value rather setting the value.